test/cocotb/osdtestlib/eval_utility.py

```diff
@@ -4,59 +4,87 @@
 from cocotb.result import TestFailure
 from cocotb.triggers import RisingEdge
 
+types = ['REG', 'PLAIN', 'EVENT']
+
+subtypes = ['REQ_READ_REG_16', 'REQ_READ_REG_32', 'REQ_READ_REG_64',
+            'REQ_READ_REG_128', 'RESP_READ_REG_SUCCESS_16',
+            'RESP_READ_REG_SUCCESS_32', 'RESP_READ_REG_SUCCESS_64',
+            'RESP_READ_REG_SUCCESS_128', 'RESP_READ_REG_ERROR',
+            'REQ_WRITE_REG_16', 'REQ_WRITE_REG_32', 'REQ_WRITE_REG_64',
+            'REQ_WRITE_REG_128', 'RESP_WRITE_REG_SUCCESS',
+            'RESP_WRITE_REG_ERROR']
+
 class EvalUtility:
     """
-    Provides an additional method to check if a register which was read out 
-    contains the correct values
+    Provides additional methods to check if a packet which was read contains the
+    correct values
     """
 
     @cocotb.coroutine
-    def evaluate_response(self, dut, response, destination, source, reg_width,
-                          expected_payload):
-        """Compare a received payload with its expectation value and output
+    def evaluate_packet(self, dut, packet, mask, destination, source, type,
+                        subtype, expected_payload):
+        """Compare a packet with its expectation value and output
            the result in the console
         Args:
             dut:                device under test.
-            response:           read response of the debug module.
+            packet:             debug interconnect packet which was returned by
+                                the module.
+            mask:               list of boolean variables indicating which flits
+                                of the payload are to be ignored
             destination:        id of the module which initially was the
                                 destination.
             source:             id of the module which initially was the source.
-            reg_width:          data width of the register the data was read 
-                                from/written to.
+            type:               2 bits identifying the packet as a
+                                register access or a debug event.
+            subtype:            for reg access:    information about word width, 
+                                                   request or response, 
+                                                   reported errors.
+                                for debug packets: number of payload words.
             expected_payload:   expected received payload.
         Returns:
             -
         """
 
         result = 1
 
-        if response[0] != source:
-            # test failed if the response was intended for a different module
-            raise TestFailure("Expected dest to be 0x%x, got 0x%x" %
-                              (source, response[0]))
-            result = 0
+        # the destination header is only available in packets using the new
+        # format
+        if packet.new:
```

Comment:
If you find code using the old packet format, open an issue on github for me to look at, or fix the Verilog code yourself. All work in this branch is geared towards the new format, we don't need to introduce costly workarounds.